docs/installation.md

```diff
@@ -0,0 +1,289 @@
+# Installation and Usage
+
+## Dependencies
+
+**Python (required for code generation)**
```

Comment:
Python is the only dependency that should be noted. It is assumed that users will have the required dependency for there languages.